Makefile

```diff
@@ -1,12 +1,30 @@
+.PHONY: install install/observability-core install/observability-edge test-acm-%-deploy
 
-install: install/observability-core
+install: install/observability-core install/observability-edge
 
 # Setup the secret required for observability then generate and apply the files
-install/observability-core:
+install/observability-core: acm/odh-core/acm-observability/kustomization.yaml
 	DOCKER_CONFIG_JSON=`oc extract secret/pull-secret -n openshift-config --to=-`
 	oc create secret generic multiclusterhub-operator-pull-secret -n open-cluster-management-observability --from-literal=.dockerconfigjson="$$DOCKER_CONFIG_JSON" --type=kubernetes.io/dockerconfigjson
 	kustomize build acm/odh-core/acm-observability | kubectl apply -f
 
 # Create the whitelist for model metrics
-install/observability-edge:
-  kustomize build acm/odh-edge/acm-observability | kubectl apply -f
\ No newline at end of file
+install/observability-edge: acm/odh-edge/acm-observability/kustomization.yaml
+	kustomize build acm/odh-edge/acm-observability | kubectl apply -f
+
+# Deploy app using ACM GitOps flow, from custom GitHub org/branch, to custom namespace
+# Example invocations:
+# make -e GIT_REPO_URL="https\://github.com/opendatahub-io/ai-edge" -e GIT_BRANCH=my-git-branch -e TEST_NAMESPACE=my-test-namespace test-acm-bike-rental-app-deploy
+# make -e GIT_REPO_URL="https\://github.com/opendatahub-io/ai-edge" -e GIT_BRANCH=my-git-branch -e TEST_NAMESPACE=my-test-namespace test-acm-tensorflow-housing-app-deploy
+test-acm-%-deploy: test/acm/%/kustomization.yaml
+ifndef GIT_REPO_URL
+	$(error GIT_REPO_URL is undefined)
+endif
+ifndef GIT_BRANCH
+	$(error GIT_BRANCH is undefined)
+endif
+ifndef TEST_NAMESPACE
+	$(error TEST_NAMESPACE is undefined)
+endif
+	kustomize build test/acm/$(subst -deploy,,$(subst test-acm-,,$@))/ | sed -e "s|https://github.com/opendatahub-io/ai-edge|$(GIT_REPO_URL)|g" -e "s|my-git-branch|$(GIT_BRANCH)|g" -e "s|my-test-namespace|$(TEST_NAMESPACE)|g" | oc apply -f - --dry-run=client
+	kustomize build test/acm/$(subst -deploy,,$(subst test-acm-,,$@))/ | sed -e "s|https://github.com/opendatahub-io/ai-edge|$(GIT_REPO_URL)|g" -e "s|my-git-branch|$(GIT_BRANCH)|g" -e "s|my-test-namespace|$(TEST_NAMESPACE)|g" | oc apply -f -
```

Comment:
```suggestion
# Generate app using ACM GitOps flow, from custom GitHub org/branch, to custom namespace
# Example invocations:
# make -e GIT_REPO_URL="https\://github.com/opendatahub-io/ai-edge" -e GIT_BRANCH=my-git-branch -e TEST_NAMESPACE=my-test-namespace test-acm-bike-rental-app-generate
# make -e GIT_REPO_URL="https\://github.com/opendatahub-io/ai-edge" -e GIT_BRANCH=my-git-branch -e TEST_NAMESPACE=my-test-namespace test-acm-tensorflow-housing-app-generate
test-acm-%-generate: test/acm/%/kustomization.yaml
ifndef GIT_REPO_URL
	$(error GIT_REPO_URL is undefined)
endif
ifndef GIT_BRANCH
	$(error GIT_BRANCH is undefined)
endif
ifndef TEST_NAMESPACE
	$(error TEST_NAMESPACE is undefined)
endif
	kustomize build test/acm/$(subst -generate,,$(subst test-acm-,,$@))/ | sed -e "s|___REPO___|$(GIT_REPO_URL)|g" -e "s|___BRANCH___|$(GIT_BRANCH)|g" -e "s|___NAMESPACE___|$(TEST_NAMESPACE)|g"

```
1. I think we should omit the oc apply from the make target. Allowing the user to first view the generated yaml is more robust and they can then apply it. We can either do that by generating the yaml in a file or to generating it to stdout, then we'd add in the README to do an oc apply. I tried the stdout approach locally so my suggestions here are based on that.
2. IMHO using placeholders instead of default values is also more robust specially if we want to allow for adding "their own test applications" , that way it's clear in the `test/acm/` manifests that they're placeholders without the need to read the Makefile.